AsyncDisplayKit/Layout/ASStackLayoutSpec.mm

```diff
@@ -23,37 +23,104 @@
+                                     spacing:(CGFloat)spacing
+                        contentJustification:(ASStackLayoutJustifyContent)justifyContent
+                               itemAlignment:(ASStackLayoutAlignItems)alignItems
+                                    children:(NSArray *)children
+{
+  return [[self alloc] initWithDirection:direction
+                                 spacing:spacing
+                    contentJustification:justifyContent
+                           itemAlignment:alignItems
+                                children:children];
+}
+
+- (instancetype)initWithDirection:(ASStackLayoutDirection)direction
+                          spacing:(CGFloat)spacing
+             contentJustification:(ASStackLayoutJustifyContent)justifyContent
+                    itemAlignment:(ASStackLayoutAlignItems)alignItems
+                         children:(NSArray *)children;
+{
+  self = [super init];
+  if (self) {
+    _direction = direction;
+    _alignItems = alignItems;
+    _spacing = spacing;
+    _justifyContent = justifyContent;
+    
+    _children = std::vector<id<ASLayoutable>>();
     for (id<ASLayoutable> child in children) {
-      spec->_children.push_back(child);
+      _children.push_back(child);
     }
   }
-  return spec;
+  return self;
+}
+
+- (void)addChild:(id<ASLayoutable>)child
+{
+  ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
+  _children.push_back(child);
+}
+
+- (void)addChildren:(NSArray *)children
+{
+  for (id<ASLayoutable> child in children) {
+    [self addChild:child];
+  }
+}
+
+
+- (void)setDirection:(ASStackLayoutDirection)direction
+{
+  ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
+  _direction = direction;
+}
+
+- (void)setAlignItems:(ASStackLayoutAlignItems)alignItems
+{
+  ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
+  _alignItems = alignItems;
+}
+
+- (void)setJustifyContent:(ASStackLayoutJustifyContent)justifyContent
+{
+  ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
+  _justifyContent = justifyContent;
 }
 
-+ (instancetype)new
+- (void)setSpacing:(CGFloat)spacing
 {
-  ASDISPLAYNODE_NOT_DESIGNATED_INITIALIZER();
+  ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
+  _spacing = spacing;
 }
 
 - (ASLayout *)measureWithSizeRange:(ASSizeRange)constrainedSize
 {
-  const auto unpositionedLayout = ASStackUnpositionedLayout::compute(_children, _style, constrainedSize);
-  const auto positionedLayout = ASStackPositionedLayout::compute(unpositionedLayout, _style, constrainedSize);
-  const CGSize finalSize = directionSize(_style.direction, unpositionedLayout.stackDimensionSum, positionedLayout.crossSize);
+  ASStackLayoutSpecStyle style = {.direction = _direction,
+    .spacing = _spacing,
```

Comment:
Align with indentation, or this one looks like a good candidate for single-lining